packages/contracts/test/contracts/discovery/Batch.ts

```diff
@@ -0,0 +1,49 @@
+describe("Batch", () => {
+  describe("constructor", () => {
+    it("works with valid arguments");
+    it("fails with an empty list of projects");
+    it("fails with an address that's not a project");
+    it("fails if one of the projects is not whitelisted");
+  });
+
+  describe("vote", () => {
+    it("accepts votes for projects");
+    it("does not accept two votes from the same user in the same project");
+    it(
+      "does not accept votes from people who exhausted their votes for this batch"
+    );
+    it("allows including an investment in the peoples pool");
+    it("allows including an investment in the stakers pool");
+    it("fails to an investment in the peoples pool for non-eligible users");
+  });
+
+  describe("getWinners", () => {
+    // assume a batch competing for 2 slots over 2 weeks
+    it("includes 1 winner after 1 week of votes");
+    it("includes 2 winners after 2 weeks of votes");
+    it("has no winners if no votes are received");
+  });
+
+  describe("votesPerProject", () => {
+    it("counts 1 vote per project, linearly");
+  });
+
+  describe("weightedVotesPerProject", () => {
+    // this will be used for tiebreaks
+    it("gives more weight to early votes, following a linear curve");
```

Comment:
How do you propose we do this? A multipler on the difference between end of the voting period and the current timestamp?

Reply:
there needs to be some maximum multiplier, given at the constructor. say it's 5% bonus. if you vote right at the start, you get 1.05 * 10^18
if you vote at the end, you get 1 * 10^18

(multiplying by 10^18) is the usual approach because we can't represent decimal numbers

Reply:
Is that bonus a constant or set by the controller?